build.gradle

```diff
@@ -1,67 +1,134 @@
-buildscript
-{
-    repositories 
-    {
-        jcenter()
-        maven { url = "http://files.minecraftforge.net/maven" }
+buildscript {
+    repositories {
+        maven { url = 'https://maven.minecraftforge.net/' }
+        mavenCentral()
     }
-    
-    dependencies 
-    {
-        classpath 'net.minecraftforge.gradle:ForgeGradle:2.3-SNAPSHOT'
+    dependencies {
+        classpath 'net.minecraftforge.gradle:ForgeGradle:5.+'
     }
 }
 
-apply plugin: 'net.minecraftforge.gradle.forge'
+apply plugin: 'net.minecraftforge.gradle'
+// Only edit below this line, the above code adds and enables the necessary things for Forge to be setup.
+apply plugin: 'eclipse'
+apply plugin: 'maven-publish'
+
 
-if (project.hasProperty("dev")) 
+if (project.hasProperty("dev"))
 {
     version += "-" + dev
 }
 
-group = "mchorse.mappet"
-archivesBaseName = "mappet"
-sourceCompatibility = targetCompatibility = "1.8"
+group = 'mchorse.mappet' // http://maven.apache.org/guides/mini/guide-naming-conventions.html
+archivesBaseName = 'mappet'
+sourceCompatibility = targetCompatibility = compileJava.sourceCompatibility = compileJava.targetCompatibility = '1.8'
+
+minecraft {
+    // The mappings can be changed at any time, and must be in the following format.
+    // snapshot_YYYYMMDD   Snapshot are built nightly.
+    // stable_#            Stables are built at the discretion of the MCP team.
+    // Use non-default mappings at your own risk. they may not always work.
+    // Simply re-run your setup task after changing the mappings to update your workspace.
+    //mappings channel: 'snapshot', version: '20171003-1.12'
+    mappings channel: 'snapshot', version: '20171003-1.12'
+    // makeObfSourceJar = false // an Srg named sources jar is made by default. uncomment this to disable.
+
+    // accessTransformer = file('src/main/resources/META-INF/accesstransformer.cfg')
+
+    // Default run configurations.
+    // These can be tweaked, removed, or duplicated as needed.
+    runs {
+        client {
+            workingDirectory project.file('run')
+
+            // Recommended logging data for a userdev environment
+            property 'forge.logging.markers', 'SCAN,REGISTRIES,REGISTRYDUMP'
+
+            // Recommended logging level for the console
+            property 'forge.logging.console.level', 'debug'
+        }
 
-compileJava {
-    sourceCompatibility = targetCompatibility = "1.8"
-}
+        server {
 
-minecraft 
-{
-    version = "1.12.2-14.23.3.2655"
-    runDir = "run"
-    
-    mappings = "snapshot_20171003"
-    
-    clientJvmArgs = ["-Xmx1G"]
-    serverJvmArgs = ["-Xmx1G"]
-
-    replace "%VERSION%", project.version
-    replace "%MCLIB%", project.mclib
```

Comment:
Removing this is not acceptable. This is required for proper building, why did you remove this?